### DISCUSSION TITLE
Required Apps - recognized node version is too high

### DISCUSSION THREAD

**Discussion by gald33** [ConsenSys-archive/truffle#5877] [Installation]
Hello, every operation I'm trying to do sends me to the Required Apps page, there I have installed all the prerequisites, only the Node and NPM versions listed as installed are too high. I run the command "nvm use 16.19.0" on the VSCode console (zsh, on macOS) and check that the versions are indeed in the specified range. However I have newer versions installed (Node 19.6.0, NPM 9.4.0) and this is what the extension recognizes. 
I don't really want to, but even if I try I am unsuccessful in "really downgrading" the machine's Node's version.

How can I make the extension work?

**Comment by cipaterson** [comment]
Hi @gald33,
most likely your problem is to do with how nvm remembers the "use" version.  AFAIK, run `nvm use 16.19.0` in a shell would change the node version only in that shell.  Try `nvm list` and you may see that the local shell's version (->) is different to the default version (an alias that nvm uses to set the version for new shells).  Also using `nvm use` in a vscode terminal would change the node version in that shell, perhaps not the version used by the parent vscode process that the extension runs in.

Try this:
exit vscode,
set the default node version with `nvm alias default 16.19.0`
run vscode,
open the terminal and check the nvm list then.
